### DISCUSSION TITLE
Is differentiating BLAS calls supported?

### DISCUSSION THREAD

**Discussion by roastduck** [EnzymeAD/Enzyme#1250] [Q&A] [answered]
Just a quick question: is differentiating BLAS calls supported? I found some code related to BLAS, but I didn't find any documentation.

**Comment by tgymnich** [comment]
Yes some level 1 functions like, `dot`, `axpy`, `copy`, `asum`,   and `scal` should be supported. @ZuseZ4  is working on extending our BLAS support https://github.com/EnzymeAD/Enzyme/pull/1208.

See: https://github.com/wsmoses/Enzyme/blob/4f95c0ff247dccde5cf1669be1c8bfa39831b851/enzyme/Enzyme/BlasDerivatives.td

**Comment by ZuseZ4** [comment, marked as answer]
There isn't any documentation right now, because I'm just writing it, once I'll finished all I will add some notes do the docs.
For the meantime I would not recommend using Blas+Enzyme for the C++ side. Unlike Enzyme.jl we don't have any fallback for not yet implemented functions. Thus if you start using any of the interesting functions it is pretty likely to fail. 

I am currently testing quite a few new functions, once that's done we should hopefully have
scal, axpy, dot, gemv, ger, and gemm. Thanks to Tablegen other functions shouldn't take too long to add after that.
Are there any specific ones which you need?
